### DISCUSSION TITLE
iOS Version

### DISCUSSION THREAD

**Discussion by KintCark** [ShiftHackZ/Stable-Diffusion-KMP#357] [Ideas]
Is it possible to get this app on iOS?

**Comment by ShiftHackZ** [comment]
App development for iOS is more complicated, so to get this app on iOS, IMHO we need to complete at least this action items:

- Make a deep investigation if all of app features can be implemented on iOS (I think LocalDiffusion generation providers may not be possible, but I may be wrong here).
- Find an iOS developer who is interested in implementing and maintaining the same code-base for iOS (I am not iOS developer myself).
- Need to find funding for Apple hardware (Mac computer and iPhone) for the iOS developer, because iOS development is only possible on Mac and debugging this kind of app most likely will require real phone. 
- Need to find funding at least $100/per year to pay for Apple Development program fee.

To sum up: for the small non-profit app which is highly specialized and has no monetization model, I don't believe it is possible.

But if there is anyone interested in creating the same iOS app - the code of this app is open source, and can be freely re-used if the new app will also be open source and published with the compliant open source license.

**Comment by ShiftHackZ** [comment]
Update - there is a technical possibility to refactor current codebase to Kotlin Multiplatform and maintain Android and iOS apps from a single codebase. Working on it.

**Comment by ShiftHackZ** [comment]
Closing this discussion, as SDAI is now available on the Apple AppStore: https://apps.apple.com/us/app/sdai-ai-image-generator/id6778314183

<img width="1280" height="720" alt="telegram-cloud-photo-size-2-5300937263833356975-y" src="https://github.com/user-attachments/assets/751f57db-1077-4992-b48d-7f5da573b706" />
